Add start selected program feature to Home Connect documentation (#43784)
home-assistant/home-assistant.github.io: source/_integrations/home_connect.markdown

### Action: Set program and options

> The `home_connect.set_program_and_options` action starts or selects a program or starts a program that it is already selected. 

Reviewer: ```suggestion
The `home_connect.set_program_and_options` action selects a program or starts a program that is already selected. 
```

Author: This wasn't an error; there are now three different modes of working:

- Start a program
- Select a program
- Start a program that it is already selected on the appliance

Reviewer: oh, I misunderstood. But how can you start a program if no program is selected?

Author: You would start a program using the program key and optionally the different option keys,
The API allows to start a program and select a program using two different endpoints, but I just added a feature that reads the selected program and selected options, overrides those with the data given in the service call and calls the start endpoint.
Is an attempt to imitate the functionality that the app already has where you can start something that already has been selected from, for example, the appliance manually.

Reviewer: I see... I rephrased it a bit and added some suggestions. Is this accurate?

{% note %}

> If the `program` attribute is not set, this action sets the options for the active or the selected program.

Reviewer: ```suggestion
If the `program` attribute is not set, this action sets the options for the active or selected program instead of starting or selecting a new one.
```

{% endnote %}
{% note %}

> When "Start selected" is chosen, the selected program and options are used.

Reviewer: ```suggestion
When **Start selected** is chosen, the selected program and options are used.
```

> When **Start selected** is chosen, the selected program and options are used.

Reviewer: ```suggestion
When **Start selected** is chosen, the action uses the program and options currently selected on the appliance. If you also provide a program value, it overrides the selected program. Any option values you provide, override matching options, or are added as new options, before the appliance starts.
```

Author: This is much better! thank you :)

> If program or option values are provided in this call, program overrides the selected program, matching options are overridden, and new options are added before start.

Reviewer: ```suggestion
If program or option values are provided, the program overrides the selected program, matching options are overridden, and new options are added before start.
```

Author: ok, but I'm going to change "the program" to "the **program** key" to represent that it is the option key from the service
